src/libs/CardUtils.ts

```diff
@@ -135,14 +135,21 @@ function maskCard(lastFour = ''): string {
  * Converts given 'X' to '•' for the entire card string.
  *
  * @param cardName - card name with XXXX in the middle.
+ * @param bank - card bank.
  * @returns - The masked card string.
  */
-function maskCardNumber(cardName: string): string {
-    if (!cardName || cardName === '') {
+function maskCardNumber(cardName: string, bank: string | undefined): string {
+    if (!cardName || cardName === '' || !bank) {
         return '';
     }
     const hasSpace = /\s/.test(cardName);
     const maskedString = cardName.replace(/X/g, '•');
+    const isAmexBank = [CONST.COMPANY_CARD.FEED_BANK_NAME.AMEX, CONST.COMPANY_CARD.FEED_BANK_NAME.AMEX_DIRECT].some((value) => value === bank);
+
+    if (isAmexBank && maskedString.length === 15) {
+        return maskedString.replace(/(.{4})(.{6})(.{5})/, '$1 $2 $3');
```

Comment:
Coming from https://expensify.slack.com/archives/C06ML6X0W9L/p1731922953677169?thread_ts=1731525071.415359&cid=C06ML6X0W9L

@koko57  we should early return cardName if hasSpace is true